pages/a/ms-azure-guest-post.html

```diff
@@ -0,0 +1,89 @@
+title: Crate guest post for Azure - Setting up Azure for Crate
+author: Chris Ward
+description: tbc
+created: 2016-02-04
+status: publish
+post_type: post
+tags: a, b
+category: developernews
+
+# Crate guest post for Azure - Setting up Azure for Crate
+
+Crate.io has been striving to create the most scalable database in our industry. We aim to make scaling your data from millions to billions of records and beyond, as simple as possible.
+
+We have clients who write and query gigabytes of data a minute, store terabytes a day and love the performance and stability that Crate offers. But the question always remained at the back of our minds, when taken to the extreme, what is Crate really capable of?
+
+Our aims were to:
+
+- See how Crate would distribute shards and partitions across such a large cluster
+- Examine the overhead added by a the cluster management
+- See what speeds the cluster could write data at
+- Another
+- And another
+
+Microsoft Azure were generously lent us the use of 8000 cores as the infrastructure for this test and engineer time to guide us through problems and questions we might have.
+
+Microsoft hoped to prove that their Azure platform was reliable enough for a project of such rapid scaling and resource intensive.
+
+So, what happened and how did we do it?
+
+We aimed to create a 1001 node Crate cluster and xxx. Setting up such a cluster on any kind of infrastructure requires planning and whilst cloud hosts simplify this process, understanding how to translate your requirements into their paradigms requires research and experimentation.
+
+Azure has several concepts that we needed to enable and configure to get our cluster started:
+
```

Comment:
* Resource [P]
  * Deployment - a set of (referencing and depending) resources that should be deployed on the platform
* Network [P]
  * VirtualNetwork [R]
    * Subnet (property of VirtualNetwork)
  * NetworkInterface [R]
     IPConfiguration (property of NetworkInterface)
  * PublicIPAddress [R]
  * NetworkSecurityGroup [R]
    * Rules (inbound and outbound, property of NSG)
* Storage [P]
  * StorageAccount [R]
* Compute [P]
  * AvailabilitySet \[R\] (make sure vms are scheduled on correct hosts, [read](https://azure.microsoft.com/en-us/documentation/articles/virtual-machines-manage-availability/#avoid-single-instance-virtual-machines-in-availability-sets))
  * VirtualMachine [R]
  * VirtualMachineExtension [R]

[P] are Providers (you can list them using the cli `azure provider list`)
[R] are Resources